config/default/test/tasks/pregnancy_home_visit.spec.js

```diff
@@ -0,0 +1,179 @@
+const chai = require('chai');
+const expect = chai.expect;
+const path = require('path');
+const moment = require('moment');
+const sinon = require('sinon');
+const TestRunner = require('medic-conf-test-harness');
+const { MAX_DAYS_IN_PREGNANCY, getRangeForTask, getTaskTestDays } = require('../test-helpers');
+const { pregnancyRegistrationScenarios, pregnancyHomeVisitScenarios } = require('../form-inputs');
+const harness = new TestRunner({
+  xformFolderPath: path.join(__dirname, '../../forms/app'),
+});
+let clock;
+const TEST_INTERVAL_DAYS = 7;
+
+describe('pregnancy home visit tests', () => {
+  before(async () => {
+    return await harness.start();
+  });
+  after(async () => {
+    return await harness.stop();
+  });
+  beforeEach(async () => {
+    await harness.clear();
+    //await harness.setNow(now);
+    //await harness.flush(1);
+    return await harness.loadForm('pregnancy');
+  });
+  afterEach(() => {
+    expect(harness.consoleErrors).to.be.empty;
+    if (clock) clock.restore();
+  });
+
+  //ANC Home Visit: 12, 20, 26, 30, 34, 36, 38, 40 weeks (Known LMP)
+  //show from 7 days before due date, show until 14 days after due date
+  //Register date: 1999-10-10
+  //LMP: 1999-08-01
+  //LMP+12 weeks: 1999-10-24 (From: 1999-10-17, To: 1999-11-07)
+  //LMP+20 weeks: 1999-12-19 (From: 1999-12-12, To: 2000-01-02)
+
+  const pregnancyHomeVisitTaskFirst = {
+    pre: 7,
+    post: 14,
+    triggers: [12, 20, 26, 30],
+    triggerUnit: 'weeks'
+  };
+
+  const pregnancyHomeVisitTaskSecond = {
+    pre: 6,
+    post: 7,
+    triggers: [34, 36, 38, 40],
+    triggerUnit: 'weeks'
+  };
+
+  const pregnancyHomeVisitUnknownLMPTask = {
+    pre: 6,
+    post: 7,
+    triggers: Array(21).fill().map((item, index) => (1 + index) * 2),//[2, 4, 6, ..., 40, 42]
+    triggerUnit: 'weeks'
+  };
+
+  const pregnancyHomeVisitTaskDays = getRangeForTask(pregnancyHomeVisitTaskFirst).concat(getRangeForTask(pregnancyHomeVisitTaskSecond));
+
+  const pregnancyHomeVisitUnknownLMPTaskDays = getRangeForTask(pregnancyHomeVisitUnknownLMPTask);
+
+  it('pregnancy home visit task should show', async () => {
+    await harness.setNow('1999-10-10');//10 weeks after LMP date
+    const pregnancy = await harness.fillForm('pregnancy', ...pregnancyRegistrationScenarios.safe10Weeks);
+    expect(pregnancy.errors).to.be.empty;
+    for (const day of getTaskTestDays(70, 32 * 7, pregnancyHomeVisitTaskFirst, TEST_INTERVAL_DAYS)
+      .concat(getTaskTestDays(32 * 7 + 1, MAX_DAYS_IN_PREGNANCY, pregnancyHomeVisitTaskSecond, TEST_INTERVAL_DAYS))) {
+      await harness.setNow('1999-08-01');//10 weeks after LMP date
+      await harness.flush(day);
+      const taskForHomeVisit = await harness.getTasks({ title: 'task.anc.pregnancy_home_visit.title' });
+      if (pregnancyHomeVisitTaskDays.includes(day)) {
+        expect(taskForHomeVisit.length).to.be.above(0, day);
+        //Getting two tasks with same title at day 245 because the window overlaps
```

Comment:
These bugs resulting from window-overlap are super annoying. In general, they can be serious problems impacting treatment because one follow-up on the day the windows overlap can result in two tasks getting result. I'd recommend making adjustments so these windows don't overlap.

I also thought it would be great to just add a feature to the harness which asserts any time who events overlap. What would you think of that feature?

Reply:
I'd love that feature in harness. In this case, we have avoided overlap by changing the time window but I missed to update the test and comment. Will update it. Thanks.

Reply:
https://github.com/medic/medic-conf-test-harness/issues/5